inbm/dispatcher-agent/dispatcher/source/ubuntu_source_manager.py

```diff
@@ -98,11 +102,19 @@ def __init__(self) -> None:
 
     def add(self, parameters: ApplicationAddSourceParameters) -> None:
         """Adds a source file and optional GPG key to be used during Ubuntu application updates."""
-        # Step 1: Add key (Optional)
+        # Step 1: Verify gpg key uri from trusted repo list 
+        try:
+            url = parameters.gpg_key_uri
+            #URL slicing to remove the last segment (filename) from the URL 
+            source = url.value[:-(len(url.value.split('/')[-1]) + 1)]
```

Comment:
Consider handling the case where `parameters.gpg_key_uri` is `None` before attempting to slice the URL, as this could lead to an AttributeError if not checked. [important]